packages/core/src/agent/message-list/prompt/convert-to-mastra-v1.ts

```diff
@@ -252,95 +252,33 @@ export function convertToV1Messages(messages: Array<MastraDBMessage>) {
-                  ...fields,
-                  type: 'tool-call',
-                  content: [
-                    ...stepInvocations.map(({ toolCallId, toolName, args }) => ({
-                      type: 'tool-call' as const,
-                      toolCallId,
-                      toolName,
-                      args,
-                    })),
-                  ],
-                });
-
-                // Only create tool-result message if there are actual results
-                const invocationsWithResults = stepInvocations.filter(ti => ti.state === 'result' && 'result' in ti);
-
-                if (invocationsWithResults.length > 0) {
-                  pushOrCombine({
-                    role: 'tool',
-                    ...fields,
-                    type: 'tool-result',
-                    content: invocationsWithResults.map((toolInvocation): ToolResultPart => {
-                      const { toolCallId, toolName, result } = toolInvocation;
-                      return {
-                        type: 'tool-result',
-                        toolCallId,
-                        toolName,
-                        result,
-                      };
-                    }),
-                  });
-                }
-              }
-            }
-          }
+          // Tool invocations are now only stored in parts array
+          // No need to check for separate toolInvocations field
 
           break;
         }
 
-        const toolInvocations = message.content.toolInvocations;
+        // Extract tool invocations from parts
+        const partsArray = (message.content as any).parts || [];
+        const toolInvocationParts = partsArray.filter(
+          (p: any) => p.type === 'tool-invocation' && p.toolInvocation?.toolName !== 'updateWorkingMemory',
+        ) as any[];
+
+        // Extract text content from parts
+        const textContent = partsArray.find((p: any) => p.type === 'text')?.text || '';
 
-        if (toolInvocations == null || toolInvocations.length === 0) {
-          pushOrCombine({ role: 'assistant', ...fields, content: content || '', type: 'text' });
+        if (toolInvocationParts.length === 0) {
+          pushOrCombine({ role: 'assistant', ...fields, content: textContent, type: 'text' });
           break;
         }
 
-        const maxStep = toolInvocations.reduce((max, toolInvocation) => {
-          return Math.max(max, toolInvocation.step ?? 0);
+        const maxStep = toolInvocationParts.reduce((max: number, part: any) => {
+          const step = part.toolInvocation?.step ?? 0;
+          return Math.max(max, step);
         }, 0);
 
         for (let i = 0; i <= maxStep; i++) {
-          const stepInvocations = toolInvocations.filter(
-            toolInvocation => (toolInvocation.step ?? 0) === i && toolInvocation.toolName !== 'updateWorkingMemory',
-          );
+          const stepInvocations = toolInvocationParts.filter((part: any) => (part.toolInvocation?.step ?? 0) === i);
```

Comment:
no any and why remove the working memory check? seems like it was probably there for a reason